### PR TITLE
parameterise the GeodesyMLValidator class

### PR DESCRIPTION
**Pull request by heyagosper**
class path resources injected by Spring in WebAppConfig

Unit test updated to pass dummy args to constructor  just to make it compile. Still need to fix this so the test passes.

### CODE REVIEW

File: src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java [hunk 1]
```diff
@@ -1,31 +1,31 @@
 package au.gov.ga.geodesy.port.adapter.geodesyml;
 
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.util.List;
+import au.gov.ga.xmlschemer.SchemaValidator;
+import au.gov.ga.xmlschemer.Violation;
+import org.xml.sax.SAXException;
```
Comment (lbodor) [inline src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java]: Your intellij is configured to use different convention for ordering imports. Could you please reconfigure it to match the convention used by eclipse (http://stackoverflow.com/questions/14716283/is-it-possible-for-intellij-to-organize-imports-the-same-way-as-in-eclipse). Let's use the eclipse ordering until we decide to change it.

File: src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
     private SchemaValidator schemaValidator;
 
-    public GeodesyMLValidator() {
+    /**
+     * Constructor, takes the root level XSD source and the path to the catalog file.
+     * @param xsdSource
+     * @param catalogPath
+     */
+    public GeodesyMLValidator(Source xsdSource, String catalogPath) {
```
Comment (lbodor) [inline src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java]: Was passing in the xsd source necessary? With the schema passed in externally, GeodesyMLValidator adds nothing to the generic SchemaValidator.

File: src/test/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidatorTest.java [hunk 3]
```diff
@@ -1,26 +1,54 @@
...
 
-    private GeodesyMLValidator validator = new GeodesyMLValidator();
+    private GeodesyMLValidator geodesyMLValidator = null;
+
+    @BeforeTest
+    public void setup() {
+        ClassPathXmlApplicationContext context = null;
+        try {
+            context = new ClassPathXmlApplicationContext();
+            Resource catalogResource = context.getResource("classpath:xsd/geodesyml-1.0.0-SNAPSHOT/catalog.xml");
```
Comment (lbodor) [inline src/test/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidatorTest.java]: In this class, the established method for loading resources from classpath is using ResourceUtils, what is your reason for deviating?

File: src/test/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidatorTest.java [hunk 4]
```diff
@@ -1,26 +1,54 @@
...
     public void testSchemaValidation() throws Exception {
         StreamSource xml = new StreamSource(ResourceUtils.getFile("classpath:sitelog/geodesyml/MOBS.xml"));
-        List<Violation> violations = validator.validate(xml);
+        List<Violation> violations = geodesyMLValidator.validate(xml);
         assertNoViolations(violations);
     }
 
+    @Test
+    public void testSchemaValidationFailure() throws Exception {
+        // load a file with 2 known schema violations
```
Comment (lbodor) [inline src/test/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidatorTest.java]: Is the code below not self-documenting? How long did it take you to work out what it did and what the comment should be?

File: src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java (hunk 2 above)
Comment (heyagosper) [inline src/main/java/au/gov/ga/geodesy/port/adapter/geodesyml/GeodesyMLValidator.java]: Well, GeodesyMLValidator is a spring bean ready for autowiring that has all the resources it needs to pass to the schema validator pre-configured. So it's a convenience...

**Pull request merged**